src/dev/impl/DevToys/ViewModels/Tools/Graphic/ImageConverter/ImageConversionWorkItem.cs

```diff
@@ -0,0 +1,133 @@
+
+        internal ImageConversionWorkItem(StorageFile file)
+        {
+            Arguments.NotNull(file, nameof(file));
+
+            FileName = file.Name;
+            FilePath = file.Path;
+
+            DeleteCommand = new RelayCommand(ExecuteDeleteCommand);
+            SaveCommand = new AsyncRelayCommand<ImageConverterToolViewModel>(ExecuteSaveCommandAsync);
+
+            ComputePropertiesAsync(file).ForgetSafely(_errorHandler);
+            DecodeImageAsync(file).ForgetSafely(_errorHandler);
+        }
+
+        #region DeleteCommand
+
+        public IRelayCommand DeleteCommand { get; }
+
+        private void ExecuteDeleteCommand()
+        {
+            DeleteItemRequested?.Invoke(this, EventArgs.Empty);
+        }
+
+        #endregion
+
+        #region SaveCommand
+
+        public IAsyncRelayCommand SaveCommand { get; }
+
+        private async Task ExecuteSaveCommandAsync(ImageConverterToolViewModel viewModel)
+        {
+            string? fileExtension = ImageHelper.GetExtension(viewModel.ConvertedFormat);
+
+            var savePicker = new FileSavePicker
+            {
+                SuggestedStartLocation = PickerLocationId.ComputerFolder
+            };
+
+            savePicker.FileTypeChoices.Add(
+                fileExtension.Replace(".", string.Empty).ToUpperInvariant(),
+                new List<string>() { fileExtension!.ToLowerInvariant() });
+
+            StorageFile? newFile = await savePicker.PickSaveFileAsync();
+
+            if (newFile is not null)
+            {
+                using (IRandomAccessStream outputStream = await newFile.OpenAsync(FileAccessMode.ReadWrite))
+                {
+                    BitmapEncoder? encoder = await ImageHelper.GetEncoderAsync(viewModel.ConvertedFormat, outputStream);
+
+                    encoder.SetSoftwareBitmap(Bitmap);
+                    await encoder.FlushAsync();
+                }
+
+                DeleteCommand.Execute(this);
+            }
+        }
+
+        #endregion
+
+        private static void HandleError(Exception e)
+        {
+            throw e;
+        }
+
+        private async Task ComputePropertiesAsync(StorageFile file)
+        {
+            await TaskScheduler.Default;
+
+            var storageFileSize = (await file.GetBasicPropertiesAsync()).Size;
+            var fileSize = StorageFileHelper.HumanizeFileSize(storageFileSize, Strings.FileSizeDisplay);
+            await ThreadHelper.RunOnUIThreadAsync(() => FileSize = fileSize);
+        }
+
+        private async Task DecodeImageAsync(StorageFile file)
+        {
+            await TaskScheduler.Default;
+
+            using (IRandomAccessStream stream = await file.OpenAsync(FileAccessMode.Read))
+            {
+                BitmapDecoder decoder = await BitmapDecoder.CreateAsync(stream);
+                var bitmap = await decoder.GetSoftwareBitmapAsync();
+                await ThreadHelper.RunOnUIThreadAsync(() => Bitmap = bitmap);
```

Comment:
As mentioned in my other comment, we never dispose `bitmap`, which leads to a memory leak.

Reply:
Being disposed now so we don´t have to wait the garbage collector